Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp

```diff
@@ -2541,6 +2541,10 @@ static Bytecode::CodeGenerationErrorOr<void> for_in_of_body_evaluation(Bytecode:
     auto iterator_register = generator.allocate_register();
     generator.emit<Bytecode::Op::Store>(iterator_register);
 
+    auto result_register = generator.allocate_register();
+    generator.emit<Bytecode::Op::LoadImmediate>(js_undefined());
+    generator.emit<Bytecode::Op::Store>(result_register);
```

Comment:
Is this store 100% needed?
This would break the SSA likeness we currently have and an eventual successor of #8242 would likely need that